src/saturn_engine/worker/topics/logger.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any
+from typing import AsyncGenerator
+from typing import Callable
+from typing import Optional
+
+import dataclasses
+import json
+from enum import Enum
+
+from saturn_engine.core import TopicMessage
+from saturn_engine.utils.log import getLogger
+from saturn_engine.worker.topic import Topic
+
+
+class LoggingTopic(Topic):
+    """A topic contains default loggers for error handling
+    such as ignoring the error and logging it in debug.
+    """
+
+    class Channels(Enum):
+        IGNORE = "ignore"
+        DEBUG = "debug"
+        INFO = "info"
+        WARNING = "warning"
+        ERROR = "error"
+
+    @dataclasses.dataclass
+    class Options:
+        channel: Optional[str] = None
+
+    def resolve_channel_to_fn(self) -> Callable[[Any], None]:
+        if self.channel == self.Channels.IGNORE.value:
+            return lambda *args: None
+        elif self.channel == self.Channels.DEBUG.value:
+            return self.logger.debug
+        elif self.channel == self.Channels.INFO.value:
+            return self.logger.info
+        elif self.channel == self.Channels.WARNING.value:
+            return self.logger.warning
+        elif self.channel == self.Channels.ERROR.value:
+            return self.logger.error
+        else:
+            raise ValueError(f"Unknown channel: {self.channel}")
+
+    def __init__(self, options: Options, **kwargs: object) -> None:
+        self.channel = options.channel or self.Channels.IGNORE.value
+        self.logger = getLogger(__name__, self)
```

Comment:
use the topic name in the logger name? (And don't use our own `getLogger` helper)